### ISSUE TITLE
Arrays of connectors dimensioned with enumeration could induce errors

### ISSUE THREAD

**Issue by mpradovelasco** [OpenModelica/OpenModelica#11378]
Hi, 

Although it is a minor detail, when connecting arrays of connectors dimensioned with enumerations coud be avoided that user connect individual connectors, since numerical indices are incorrect. That is:

![image](https://github.com/OpenModelica/OpenModelica/assets/114144098/0fcb70f2-ae1d-4b45-9055-09ca523e17d0)

if user select so_c[1] and sich[1] is a wrong since so_c and sich are defined as:

``` Modelica
RealInput sich[chemicals];
RealOutput so_c[chemicals]

```
and `chemicals` is an enumeration type.

System OM 1.23 dev and Windows 11.

Thanks!

**Comment by bilderbuchi** [comment]
There is a mapping between enum values and integers, starting at 1 (https://specification.modelica.org/master/class-predefined-types-and-declarations.html#conversion-of-enumeration-to-string-or-integer), so `so_c[1]` should correspond to the first entry in the `chemicals` enum.
AFAICT, using integers and enum values interchangeably when indexing into arrays works because internally the same thing happens anyway (based on the above mapping), but it's not specifically spelled out in the specification that this is intended.

**Comment by mpradovelasco** [comment]
Thank you for your analysis @bilderbuchi. I agree with it. However, OMEdit gives an error when trying to use an integer index instead of a value of the enumeration. In the previous example, OMEdit says
`[Subscript '1' has type Integer, expected type enumeration chemicals(PD, PM,SM).]`

when using [1] in the dialog box.

**Comment by bilderbuchi** [comment]
Ah, I did not pick up that this is not working, I thought it was working unexpectedly, and you were puzzled why. 😅

**Comment by perost** [comment]
Indeed, an array dimension indexed by a Boolean or an enumeration type can only be subscripted with a Boolean or enumeration value of that type (or ':'), there's no implicit conversion of Integer to enumeration values in Modelica.

**Comment by casella** [comment]
In fact, there are function to explicitly convert enumeration to integer and vice versa. Integer(e) was introduced in Modelica 2.1  SomeEnum(i) was introduced in Modelica 3.4. They have always been mandatory, though some tools were lax in implementing this requirement. OMC is now strict about it.

@perost if an array connector is enumeration-indexed, I guess the enum values should be passed through the inst-based interface. Then, we could have a drop-down menu to select either one specific value or [:], instead of the current interface. What do you think?

**Comment by casella** [comment]
OK, please coordinate with @adeas31 to implement that. This issue is probably not that frequent, but needs to be fixed if we want to stick to the standard.
